### PR TITLE
GH-37473: [MATLAB] Add support for indexing `RecordBatch` columns by `Field` name 

### LINKED ISSUE #37473: [MATLAB] Add support for indexing `RecordBatch` columns by `Field` name
### Describe the enhancement requested

Currently, `arrow.tabular.Schema` supports indexing by `Field` name. However, `arrow.tabular.RecordBatch` does not.

The ability to index columns in a `RecordBatch` by `Field` name would be a helpful usability improvement.

**Example**
```matlab
>> t = array2table(rand(3))

t =

  3×3 table

     Var1       Var2       Var3  
    _______    _______    _______

    0.96489    0.95717    0.14189
    0.15761    0.48538    0.42176
    0.97059    0.80028    0.91574

>> rb = arrow.recordBatch(t)

rb = 

Var1:   [
    0.9648885351992765,
    0.15761308167754828,
    0.9705927817606157
  ]
Var2:   [
    0.9571669482429456,
    0.4853756487228412,
    0.8002804688888001
  ]
Var3:   [
    0.14188633862721534,
    0.421761282626275,
    0.9157355251890671
  ]
 
>> rb.column("Var1")

ans = 

[
  0.9571669482429456,
  0.4853756487228412,
  0.8002804688888001
]
```

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by kevingurney**
### Rationale for this change

Currently, `arrow.tabular.Schema` supports indexing by `Field` name. However, `arrow.tabular.RecordBatch` does not.

This pull request adds the ability to index columns in a `RecordBatch` by `Field` name.

### What changes are included in this PR?

1. Added support for indexing columns in a `RecordBatch` by `Field` name via the `column` method.

**Example**
```matlab
>> recordBatch = arrow.tabular.RecordBatch.fromArrays(...
       arrow.array([1, 2, 3]), ...
       arrow.array(["A", "B", "C"]), ...
       arrow.array([true, false, true]), ...
       ColumnNames=["A", "B", "C"] ...
   )

recordBatch = 

A:   [
    1,
    2,
    3
  ]
B:   [
    "A",
    "B",
    "C"
  ]
C:   [
    true,
    false,
    true
  ]

>> recordBatch.column("B")

ans = 

[
  "A",
  "B",
  "C"
]

>> recordBatch.column("C")

ans = 

[
  true,
  false,
  true
]
``` 
2. Removed comments about vectorizing `field` method of `Schema` and `column` method of `RecordBatch`. After further consideration, we believe it would make more sense to only allow these methods to accept scalar inputs. We could revisit support for vectorization if we overload the parenthesis operator (e.g. `recordBatch(rows, columns)`)  in the future to return another `RecordBatch`/`Schema` that only includes the selected columns/fields.
3. Fixed typo in `tSchema.m`.

### Are these changes tested?

Yes.

1. Added tests for indexing by column name using the `column` method to `tRecordBatch.m`.

### Are there any user-facing changes?

Yes.

1. Users can now index `RecordBatch` columns by name using the syntax `column(name)`.

### Future Directions

1. Consider overloading parentheses-based indexing on `RecordBatch` and `Schema`. 
* Closes: #37473

### CODE REVIEW

File: matlab/src/matlab/+arrow/+tabular/RecordBatch.m [hunk 1]
```diff
@@ -54,13 +54,17 @@
...
-            % TODO: Consider vectorizing column() in the future to support
-            % extracting multiple columns at once.
-            validateattributes(idx, "int32", "scalar");
-
-            args = struct(Index=idx);
-            [proxyID, typeID] = obj.Proxy.getColumnByIndex(args);                
+            idx = index.numericOrString(idx, "int32");
+
+            if isnumeric(idx)
+                validateattributes(idx, "int32", "scalar");
```
Comment (sgilmore10) [inline matlab/src/matlab/+arrow/+tabular/RecordBatch.m:60]: You can remove this `validateattributes` call if you include an `arguments` block:

```matlab
function arrowArray = column(obj, idx)
    arguments
         obj
         idx(1, 1)
    end
    ...
end
```
That way you don't need to re-validate the type of `idx`

File: matlab/src/matlab/+arrow/+tabular/Schema.m [hunk 2]
```diff
@@ -48,14 +48,10 @@
             idx = index.numericOrString(idx, "int32");
 
             if isnumeric(idx)
-                % TODO: Consider vectorizing field() to support extracting
```
Comment (sgilmore10) [inline matlab/src/matlab/+arrow/+tabular/Schema.m:51]: Could use an `arguments` block here as well.

File: matlab/src/matlab/+arrow/+tabular/RecordBatch.m (hunk 1 above)
Comment (sgilmore10) [reply matlab/src/matlab/+arrow/+tabular/RecordBatch.m:60]: On second thought, we should enhance the `index.numericOrString` function to accept a `AcceptNonScalar` name-value pair that defaults to `true`.

File: matlab/test/arrow/tabular/tRecordBatch.m [hunk 3]
```diff
@@ -223,6 +223,169 @@ function SchemaNoSetter(tc)
...
+            testCase.verifyEqual(actual, expected);
+
+            expected = arrow.array([true, false, true]);
+            actual = recordBatch.column("   ");
+            testCase.verifyEqual(actual, expected);
+        end
+
+        function ErrorIfColumnNameDoesNotExist(testCase)
+            % Verify that an error is thrown when trying to access a column
+            % with a Field name that is not part of the Schema of the RecordBatch.
```
Comment (sgilmore10) [inline matlab/test/arrow/tabular/tRecordBatch.m:288]: `Field` -> `field`

File: matlab/test/arrow/tabular/tRecordBatch.m [hunk 4]
```diff
@@ -223,6 +223,169 @@ function SchemaNoSetter(tc)
...
+            actual = recordBatch.column("B");
+            testCase.verifyEqual(actual, expected);
+
+            expected = arrow.array([true, false, true]);
+            actual = recordBatch.column("C");
+            testCase.verifyEqual(actual, expected);
+        end
+
+        function GetColumnByNameWithEmptyString(testCase)
+            % Verify that a column whose Field name is the empty string ("")
```
Comment (sgilmore10) [inline matlab/test/arrow/tabular/tRecordBatch.m:249]: Field name -> name

File: matlab/test/arrow/tabular/tRecordBatch.m [hunk 5]
```diff
@@ -223,6 +223,169 @@ function SchemaNoSetter(tc)
                 "MATLAB:class:SetProhibited");
         end
 
+        function GetColumnByName(testCase)
+            % Verify that columns can be accessed using a field name.
```
Comment (sgilmore10) [inline matlab/test/arrow/tabular/tRecordBatch.m:227]: can be accessed using a field name -> can be accessed by name

File: matlab/test/arrow/tabular/tRecordBatch.m [hunk 6]
```diff
@@ -223,6 +223,169 @@ function SchemaNoSetter(tc)
...
+            actual = recordBatch.column(name);
+            testCase.verifyEqual(actual, expected);
+
+            % Should match the second column whose name is "B".
+            name = 'B';
+            expected = arrow.array(["A", "B", "C"]);
+            actual = recordBatch.column(name);
+            testCase.verifyEqual(actual, expected);
+
+            % Should match the third field whose name is "123".
```
Comment (sgilmore10) [inline matlab/test/arrow/tabular/tRecordBatch.m:365]: field -> column

File: matlab/test/arrow/tabular/tRecordBatch.m [hunk 7]
```diff
@@ -223,6 +223,169 @@ function SchemaNoSetter(tc)
...
+                ColumnNames=["A", "", "C"] ...
+            );
+
+            expected = arrow.array(["A", "B", "C"]);
+            actual = recordBatch.column("");
+            testCase.verifyEqual(actual, expected)
+        end
+
+        function GetColumnByNameWithWhitespace(testCase)
+            % Verify that a column whose Field name contains only whitespace
```
Comment (sgilmore10) [inline matlab/test/arrow/tabular/tRecordBatch.m:264]: field name -> name

**Review by sgilmore10 — APPROVED** [review]
Thanks for adding this!

File: matlab/test/arrow/tabular/tRecordBatch.m (hunk 3 above)
Comment (kevingurney) [reply matlab/test/arrow/tabular/tRecordBatch.m]: Fixed.

File: matlab/test/arrow/tabular/tRecordBatch.m (hunk 4 above)
Comment (kevingurney) [reply matlab/test/arrow/tabular/tRecordBatch.m]: Fixed.

File: matlab/test/arrow/tabular/tRecordBatch.m (hunk 5 above)
Comment (kevingurney) [reply matlab/test/arrow/tabular/tRecordBatch.m]: Fixed.

File: matlab/test/arrow/tabular/tRecordBatch.m (hunk 6 above)
Comment (kevingurney) [reply matlab/test/arrow/tabular/tRecordBatch.m]: Fixed.

File: matlab/test/arrow/tabular/tRecordBatch.m (hunk 7 above)
Comment (kevingurney) [reply matlab/test/arrow/tabular/tRecordBatch.m]: Fixed.

File: matlab/src/matlab/+arrow/+tabular/RecordBatch.m (hunk 1 above)
Comment (kevingurney) [reply matlab/src/matlab/+arrow/+tabular/RecordBatch.m:60]: That's a good point. I've captured this in #37477.

Comment (kevingurney) [reply matlab/src/matlab/+arrow/+tabular/RecordBatch.m:60]: Marking this as resolved.

**Pull request merged**